projenrc/api-extractor-docs-publishing.ts

```diff
@@ -0,0 +1,150 @@
+import type { Monorepo, TypeScriptWorkspace } from 'cdklabs-projen-project-types/lib/yarn';
+import { Component, github } from 'projen';
+
+export interface ApiExtractorDocsPublishingProps {
+  /**
+   * The docs stream to publish to.
+   */
+  readonly docsStream: string;
+
+  /**
+   * The role arn (or github expression) for OIDC to assume to do the actual publishing.
+   */
+  readonly roleToAssume: string;
+
+  /**
+   * The bucket name (or github expression) to publish to.
+   */
+  readonly bucketName: string;
+}
+
+export class ApiExtractorDocsPublishing extends Component {
+  private readonly github: github.GitHub;
+  private readonly props: ApiExtractorDocsPublishingProps;
+
+  constructor(project: TypeScriptWorkspace, props: ApiExtractorDocsPublishingProps) {
+    super(project);
+
+    const gh = (project.parent! as Monorepo).github;
+    if (!gh) {
+      throw new Error('This workspace does not have a GitHub instance');
+    }
+    this.github = gh;
+
+    this.props = props;
+
+    // Add a task to run api-extractor and zip the output
+    const apiExtractorDocsTask = project.addTask('api-extractor-docs', {
+      exec: [
+        // Ensure API Extractor is installed
+        'npm install -g @microsoft/api-extractor || true',
+        // Run api-extractor to generate the API model
+        // Use || true to ensure the task continues even if api-extractor reports failures
```

Comment:
This seems counter intuitive? Why continue on failure?

Reply:
`api-extractor` will still output a file when there are certain failures. Will work on fixing the underlying issue, but I did this to temporarily unblock it.

Reply:
Maybe a better way to handle this error is to check whether the file was outputted?